trieste/space.py

```diff
@@ -119,13 +114,7 @@ def sample(self, num_samples: int) -> TensorType:
         :return: ``num_samples`` i.i.d. random points, sampled uniformly, and without replacement,
             from this search space.
         """
-        num_points = self._points.shape[0]
-        if num_samples > num_points:
-            raise ValueError(
-                "Number of samples cannot be greater than the number of points"
-                f" {num_points} in discrete search space, got {num_samples}"
-            )
-
+        tf.debugging.assert_less_equal(num_samples, len(self._points))
```

Comment:
why not keeping the message? is it self explanatory enough?

Reply:
will revert